src/components/layout/SectionLayout/SectionLayout.tsx

```diff
@@ -1,32 +1,54 @@
-import { TBColor } from '@assets/defaultTheme'
 import { Box } from '@mui/material'
 import React from 'react'
 import './sectionLayout.scss'
+import { SxProps } from '@mui/system'
+import { Theme } from '@mui/material/styles'
+
+export type SectionLayoutComponent = {
+  component: React.ReactNode
+  options?: SectionLayoutComponentsOptions
+}
+
+export type SectionLayoutComponentsOptions = {
+  type?: 'top' | 'middle' | 'bottom'
+  withPadding?: boolean
+  fitContent?: boolean
+  sx?: SxProps<Theme>
+}
 
 export type SectionLayoutProps = {
-  topComponent: React.ReactNode
-  middleComponent: React.ReactNode
-  bottomComponent: React.ReactNode
-  borderColor?: TBColor
-  backgroundColor?: TBColor
+  components: SectionLayoutComponent[]
```

Comment:
How do you plan to support the actual use of it on the landing page? And where it will be used on the DApp?

Reply:
In the DApp it is used for almost every section in the home page for example. 

The changes I will then do in the landing page are very small, an example of that are the changes in the storybook stories. 
For example:
from
```
  topComponent:` topComponent(),
  middleComponent: middleComponent(),
  bottomComponent: <Stepper {...stepperArgs} />,
  borderColor: 'green',
```
to:
```
  components: [
    { component: topComponent(), options: { type: 'top' } },
    { component: middleComponent(), options: { type: 'middle' } },
    { component: <Stepper {...stepperArgs} />, options: { type: 'bottom' } },
  ],
  borderColor: colors.green,
```

Reply:
Awesome! Thanks for the answer, looks pretty strike forward